File: backend/src/main/kotlin/com/nevexis/backend/schoolManagement/messages/Message.kt
package com.nevexis.backend.schoolManagement.messages

import com.nevexis.backend.schoolManagement.users.UserView
import kotlinx.serialization.Serializable
import java.math.BigDecimal
import java.time.LocalDateTime

data class Message(
    val id: BigDecimal? = null,
    val user: UserView,
    val content: MessageContent,
    val sendOn: LocalDateTime,
    val chatId: BigDecimal,
    val readFromUserIds: List<Int>
)

@Serializable
data class MessageContent(
    val text: String? = null,
    val files: List<FileWithBase64>? = null
)

@Serializable
data class FileWithBase64(
    val name: String,
    val base64: String,
)

File: frontend/src/model/Message.ts
import {UserView} from "./User";

export interface Message {

    id: number | null

    user: UserView,

    content: MessageContent,

    sendOn: Date | null,

    chatId: number,

    readFromUserIds: number[]

}


export interface MessageContent {

    text: string | null,

    files: FileWithBase64[] | null
}

export interface FileWithBase64 {
    name: string;
    base64: string;
}